frontend/src/createStream/__tests__/CreateStreamPageContents.test.jsx

```diff
@@ -0,0 +1,361 @@
+        deviceIndex: wrapper.state().receiverDeviceIndex,
+        setDeviceIndex: wrapper.instance().setReceiverDeviceIndex,
+        channelIndex: wrapper.state().receiverChannelIndex,
+        setChannelIndex: wrapper.instance().setReceiverChannelIndex
+      };
+      expect(props).toMatchObject(expected);
+    });
+    it("Contains 1 <Forward/> component with expected props", () => {
+      const components = wrapper.find(Forward);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+      const expected = {
+        style: { fontSize: 100 }
+      };
+      expect(props).toMatchObject(expected);
+    });
+    it("Contains 1 <Button/> component with expected props", () => {
+      const components = wrapper.find(Button);
+      expect(components).toHaveLength(1);
+
+      const props = components.at(0).props();
+      const expected = {
+        variant: "contained",
+        color: "primary",
+        onClick: wrapper.instance().createStream,
+        children: "Create Stream"
+      };
+      expect(props).toMatchObject(expected);
+    });
+  });
+  describe("componentDidMount() function", () => {
+    it("calls DeviceApi.getSenders() and getReceivers()", () => {
+      const dummySenders = [new DeviceInfo("Beep")];
+      const dummyReceivers = [new DeviceInfo("boop!")];
+
+      jest.spyOn(DeviceApi, "getSenders").mockImplementation(() => {
+        return dummySenders;
+      });
+      jest.spyOn(DeviceApi, "getReceivers").mockImplementation(() => {
+        return dummyReceivers;
+      });
+      wrapper = Enzyme.shallow(<CreateStreamPageContents />, {
+        disableLifecycleMethods: true
+      });
+      wrapper.instance().componentDidMount();
+
+      expect(DeviceApi.getSenders).toHaveBeenCalledWith(
+        wrapper.instance().setSenders
+      );
+      expect(DeviceApi.getReceivers).toHaveBeenCalledWith(
+        wrapper.instance().setReceivers
+      );
+    });
+  });
+  describe("State setting functions", () => {
+    beforeEach(() => {
+      wrapper = Enzyme.shallow(<CreateStreamPageContents />);
+    });
+    it("setSenders() function should set the state", () => {
+      const startingState = {
+        senders: [],
+        receivers: [],
+        senderDeviceIndex: -1,
+        senderChannelIndex: -1,
+        receiverDeviceIndex: -1,
+        receiverChannelIndex: -1
+      };
+      const someSenders = [new DeviceInfo("s", "lc", "f", "d", "someName")];
+      const expectedState = {
+        senders: someSenders
+      };
+
+      wrapper.setState(startingState);
+      wrapper.instance().setSenders(someSenders);
+
+      expect(wrapper.state()).toMatchObject(expectedState);
+    });
+    it("setReceivers() function should set the state", () => {
+      const startingState = {
+        senders: [],
+        receivers: [],
+        senderDeviceIndex: -1,
+        senderChannelIndex: -1,
+        receiverDeviceIndex: -1,
+        receiverChannelIndex: -1
+      };
```

Comment:
with latest changes to tests notably setSenderDeviceIndex & receiverDeviceIndex setting the corresponding ChannelIndex to -1, The values are SLIGHTLY different and so it can't really be extracted